Authenticator/Source/TableDiff.swift

```diff
@@ -31,92 +31,63 @@ protocol Identifiable {
+            changesInDiagonal[k + MAX] = changes
+            if x >= oldItems.count && y >= newItems.count {
+                return changes
             }
-        } else {
-            // The old and new rows are different, so compute the two possible change sets:
-            // one where the old row is deleted, another where the new row is inserted
-            let changesA = changesWithInsertion(from: oldRows, to: newRows,
-                comparator: rowsHaveSameIdentity, equation: rowsAreEqual)
-            let changesB = changesWithDeletion(from: oldRows, to: newRows,
-                comparator: rowsHaveSameIdentity, equation: rowsAreEqual)
-
-            // Return the shorter of the two change sets
-            return changesA.count < changesB.count ? changesA : changesB
         }
-
-    case (.Some, .None):
-        // Only old rows remain, which must be deleted
-        return changesWithDeletion(from: oldRows, to: newRows,
-            comparator: rowsHaveSameIdentity, equation: rowsAreEqual)
-
-    case (.None, .Some):
-        // Only new rows remain, which must be inserted
-        return changesWithInsertion(from: oldRows, to: newRows,
-            comparator: rowsHaveSameIdentity, equation: rowsAreEqual)
-
-    case (.None, .None):
-        // All rows are accounted for
-        return []
     }
-}
-
-private func changesWithInsertion<Row>(from oldRows: ArraySlice<Row>, to newRows: ArraySlice<Row>,
-    comparator: (Row, Row) -> Bool, equation: (Row, Row) -> Bool) -> [Change]
-{
-    let insertion = Change.Insert(index: newRows.endIndex.predecessor())
-    let changesAfterInsertion = changes(from: oldRows, to: newRows.dropLast(),
-        comparator: comparator, equation: equation)
-    return [insertion] + changesAfterInsertion
-}
-
-private func changesWithUpdate<Row>(from oldRows: ArraySlice<Row>, to newRows: ArraySlice<Row>,
-    comparator: (Row, Row) -> Bool, equation: (Row, Row) -> Bool) -> [Change]
-{
-    // TODO: Test update indices (old index or new?)
-    let update = Change.Update(index: newRows.endIndex.predecessor())
-    let changesAfterUpdate = changes(from: oldRows.dropLast(), to: newRows.dropLast(),
-        comparator: comparator, equation: equation)
-    return [update] + changesAfterUpdate
-}
-
-private func changesWithDeletion<Row>(from oldRows: ArraySlice<Row>, to newRows: ArraySlice<Row>,
-    comparator: (Row, Row) -> Bool, equation: (Row, Row) -> Bool) -> [Change]
-{
-    let deletion = Change.Delete(index: oldRows.endIndex.predecessor())
-    let changesAfterDeletion = changes(from: oldRows.dropLast(), to: newRows,
-        comparator: comparator, equation: equation)
-    return [deletion] + changesAfterDeletion
+    // With MAX = oldItems.count + newItems.count, a solution must be found by the above algorithm
+    // but here's a delete-and-insert-everything fallback, just in case.
+    return oldItems.indices.map({ .Delete(index: $0) }) + newItems.indices.map({ .Insert(index: $0) })
```

Comment:
Line Length Violation: Line should be 100 characters or less: currently 102 characters (line_length)